lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
>https://github.com/cloudant/nodejs-cloudant/blob/master/example/crud.js#L50 insert does both create and update

`nodejs-cloudant` shouldn't be determining our behaviour here; update from a LB perspective is not an alias for replace/create. I know that previously, I said that we should make it clear to users that we'll just do what the driver does, but now I'm not so sure.

What are the potential pitfalls of using `updateAttributes` as the default action here?